src/arrayMethodSort.js

```diff
@@ -4,8 +4,33 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      if (aStr < bStr) {
+        return -1;
+      }
+
+      if (aStr > bStr) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    const compare = compareFunction || defaultCompare;
```

Comment:
It's good practice to validate the `compareFunction` argument to ensure it's a function if provided. You can add a check to throw a TypeError if `compareFunction` is not a function and not undefined. This aligns with the behavior of the native `sort` method.